FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/mqtt_operations.c

```diff
@@ -0,0 +1,1069 @@
+
+        if( xTlsStatus == TLS_TRANSPORT_SUCCESS )
+        {
+            /* Connection successful. */
+            xReturnStatus = true;
+        }
+        else
+        {
+            /* Generate a random number and get back-off value (in milliseconds) for the next connection retry. */
+            xBackoffAlgStatus = BackoffAlgorithm_GetNextBackoff( &xReconnectParams, prvGenerateRandomNumber(), &usNextRetryBackOff );
+
+            if( xBackoffAlgStatus == BackoffAlgorithmRetriesExhausted )
+            {
+                LogError( ( "Connection to the broker failed, all attempts exhausted." ) );
+            }
+            else if( xBackoffAlgStatus == BackoffAlgorithmSuccess )
+            {
+                LogWarn( ( "Connection to the broker failed. Retrying connection "
+                           "after %hu ms backoff.",
+                           ( unsigned short ) usNextRetryBackOff ) );
+                vTaskDelay( pdMS_TO_TICKS( usNextRetryBackOff ) );
+            }
+        }
+    } while( ( xTlsStatus != TLS_TRANSPORT_SUCCESS ) && ( xBackoffAlgStatus == BackoffAlgorithmSuccess ) );
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static bool prvGetNextFreeIndexForOutgoingPublishes( uint8_t * pucIndex )
+{
+    bool xReturnStatus = false;
+    uint8_t ucIndex = 0;
+
+    assert( pxOutgoingPublishPackets != NULL );
+    assert( pucIndex != NULL );
+
+    for( ucIndex = 0; ucIndex < mqttopMAX_OUTGOING_PUBLISHES; ucIndex++ )
+    {
+        /* A free index is marked by invalid packet id. Check if the the index
+         * has a free slot. */
+        if( pxOutgoingPublishPackets[ ucIndex ].usPacketId == mqttopMQTT_PACKET_ID_INVALID )
+        {
+            xReturnStatus = true;
+            break;
+        }
+    }
+
+    /* Copy the available index into the output param. */
+    if( xReturnStatus == true )
+    {
+        *pucIndex = ucIndex;
+    }
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static void prvCleanupOutgoingPublishAt( uint8_t ucIndex )
+{
+    assert( pxOutgoingPublishPackets != NULL );
+    assert( ucIndex < mqttopMAX_OUTGOING_PUBLISHES );
+
+    /* Clear the outgoing publish packet. */
+    ( void ) memset( &( pxOutgoingPublishPackets[ ucIndex ] ),
+                     0x00,
+                     sizeof( pxOutgoingPublishPackets[ ucIndex ] ) );
+}
+/*-----------------------------------------------------------*/
+
+static void prvCleanupOutgoingPublishes( void )
+{
+    assert( pxOutgoingPublishPackets != NULL );
+
+    /* Clean up all the outgoing publish packets. */
+    ( void ) memset( pxOutgoingPublishPackets, 0x00, sizeof( pxOutgoingPublishPackets ) );
+}
+/*-----------------------------------------------------------*/
+
+static void prvCleanupOutgoingPublishWithPacketID( uint16_t usPacketId )
```

Comment:
should these assert calls be configASSERT calls instead?

Reply:
Yes, all demo files have been switched to use configASSERT in this commit: https://github.com/FreeRTOS/FreeRTOS/pull/724/commits/9264e2a13fe519c8eda0c08416c97f471b6e760d